src/test/kotlin/io/icure/md/client/apis/e2e/DataSampleApiImplTest.kt

```diff
@@ -220,29 +219,76 @@ internal class DataSampleApiImplTest {
             )
 
             // When
-            val attachmentDocId = testedInstance.deleteAttachment(createdDataSample.id!!, createdDocument.id!!)
+            val attachmentDocId =
+                medTechApi.dataSampleApi().deleteAttachment(createdDataSample.id!!, createdDocument.id!!)
 
             // Then
             assertEquals(createdDocument.id, attachmentDocId)
 
-            val updatedDataSample = testedInstance.getDataSample(createdDataSample.id!!)
+            val updatedDataSample = medTechApi.dataSampleApi().getDataSample(createdDataSample.id!!)
             assert(updatedDataSample.content["en"]?.documentId == null)
         }
     }
-}
\ No newline at end of file
+
+    @Test
+    @DisplayName("Create Data Sample linked to HealthElement - Success")
+    fun createDataSampleLinkedToHealthElement() {
+        runBlocking {
+            val credentials = TestUtils.UserCredentials.fromFile("pat_0857c725-3837-49ca-a3b6-f31cf7ebc61f.json")
+            val api = credentials.api
+
+            val currentUser = api.userApi().getLoggedUser()
+            val patient = api.patientApi()
+                .getPatient(currentUser.patientId ?: throw IllegalArgumentException("Test user should be a patient"))
+            val healthElement = api.healthcareElementApi()
+                .createOrModifyHealthcareElement(patient.id!!, HealthcareElement(note = "Stay hungry. Stay foolish."))
+            val dataSampleToCreate = weightDataSample().copy(
+                healthElementsIds = setOf(healthElement.id!!)
+            )
+            val createdDataSample = api.dataSampleApi().createOrModifyDataSampleFor(patient.id!!, dataSampleToCreate)
+
+            assert(createdDataSample.healthElementsIds!!.single() == healthElement.id)
+        }
+    }
+
+    @Test
+    @DisplayName("Create Data Sample and modify it to link it to HealthElement - Success")
+    fun createDataSampleAndModifyItToLinkItToHealthElement() {
+        runBlocking {
+            val credentials = TestUtils.UserCredentials.fromFile("pat_0857c725-3837-49ca-a3b6-f31cf7ebc61f.json")
+            val api = credentials.api
+
+            val currentUser = api.userApi().getLoggedUser()
+            val patient = api.patientApi()
+                .getPatient(currentUser.patientId ?: throw IllegalArgumentException("Test user should be a patient"))
+            val createdDataSample = api.dataSampleApi().createOrModifyDataSampleFor(patient.id!!, weightDataSample())
+
+            val healthElement = api.healthcareElementApi()
+                .createOrModifyHealthcareElement(patient.id!!, HealthcareElement(note = "Stay hungry. Stay foolish."))
+            val updatedDataSample = api.dataSampleApi().createOrModifyDataSampleFor(
+                patient.id!!,
+                createdDataSample.copy(modified = null, healthElementsIds = setOfNotNull(healthElement.id))
+            )
+
+            assert(updatedDataSample.healthElementsIds!!.single() == healthElement.id)
```

Comment:
As we have access to the icure-reactive-kotlin-client (through baseContactApi), that would be nice to check if contact has subcontacts properly created